subsys/bluetooth/mesh/friend.c

```diff
@@ -1202,6 +1195,56 @@ int bt_mesh_friend_init(void)
 	return 0;
 }
 
+static void update_overwrite(struct net_buf *buf)
+{
+	struct net_buf_simple_state state;
+	struct bt_mesh_ctl_friend_update *upd;
+
+	if (buf->len != 16) {
+		return;
+	}
+
+	net_buf_simple_save(&buf->b, &state);
+
+	net_buf_skip(buf, 1); /* skip IVI, NID */
+
+	if (!(net_buf_pull_u8(buf) >> 7)) {
+		goto end;
+	}
+
+	net_buf_pull(buf, 3); /* skip SEQNUM */
+
+	net_buf_skip(buf, 2) /* skip src */
+
+	net_buf_skip(buf, 2); /* skip dst */
+
+	if (TRANS_CTL_OP((u8_t *) net_buf_pull_mem(buf, 1))
+			!= TRANS_CTL_OP_FRIEND_UPDATE) {
+		goto end;
+	}
+
+	upd = net_buf_pull_mem(buf, sizeof(*upd));
+	BT_DBG("Update Previous Friend Update MD 0x%02x -> 0x01", upd->md);
+	upd->md = 1;
```

Comment:
I think this needs to be done in the friend timeout instead. There might be other messages that are enqueued after the security update, which won't be transferred after the changes in this PR. Alternatively, the update in add_update could always go out with MD=1, but it looks like that violates the spec test case.